source/Engine.cpp

```diff
@@ -1583,6 +1584,15 @@ void Engine::EnterSystem()
 
 	center = flagship->Center();
 
+	// If the player entered a system by wormhole or jump drive, center the background
+	// on the player's position. This is not done when entering a system by hyperdrive
+	// so that there is a seamless transition between systems, and entry by taking off
+	// from a planet is also excluded as to prevent the background from snapping to a
+	// different position relative to what it was when the player landed.
+	SystemEntry entry = player.GetSystemEntry();
+	if(entry == SystemEntry::WORMHOLE || entry == SystemEntry::JUMP)
+		GameData::SetBackgroundPosition(center);
```

Comment:
Technically, if the player reenters a wormhole before leaving a system, then the starfield will be the same on both sides of the wormhole. The current starfield position is also retained when loading a save.

Reply:
We'd need to retain the old position and then recognize when to use it instead of using the camera's position to handle those cases. Feels like overkill to me.